packages/node/src/subcommands/testing.service.ts

```diff
@@ -0,0 +1,253 @@
+          await this.runTest(test);
+        }),
+      );
+    }
+  }
+
+  private findAllTestFiles(dirPath: string): string[] {
+    const files: string[] = [];
+
+    readdirSync(dirPath).forEach((file) => {
+      const filePath = path.join(dirPath, file);
+      const stat = statSync(filePath);
+
+      if (stat.isDirectory()) {
+        files.push(...this.findAllTestFiles(filePath));
+      } else if (filePath.endsWith('.test.js')) {
+        files.push(filePath);
+      }
+    });
+
+    return files;
+  }
+
+  private async runTest(test: SubqlTest) {
+    logger.info(`Starting test: ${test.name}`);
+
+    // Fetch block
+    logger.debug('Fetching block');
+    const [block] = await SubstrateUtil.fetchBlocksBatches(
+      this.apiService.getApi(),
+      [test.blockHeight],
+    );
+
+    // Check filters match
+
+    // Init db
+    const schema = `test-${this.nodeConfig.subqueryName}`;
+
+    const schemas = await this.sequelize.showAllSchemas(undefined);
+    if (!(schemas as unknown as string[]).includes(schema)) {
+      await this.sequelize.createSchema(`"${schema}"`, undefined);
+    }
+
+    const modelRelations = getAllEntitiesRelations(this.project.schema);
+    await this.storeService.init(modelRelations, schema);
+    const store = this.storeService.getStore();
+
+    logger.info(JSON.stringify(test.dependentEntities));
+
+    // Init entities
+    logger.debug('Initializing entities');
+    test.dependentEntities.map(async (entity) => {
+      const attrs = entity as unknown as CreationAttributes<Model>;
+      logger.info(entity.constructor.name);
+      await store.set(`${(entity as any).name}`, entity.id, entity);
+    });
+
+    const handlerInfo = this.getHandlerInfo(test.handler);
+
+    const sandbox = new IndexerSandbox(
+      {
+        root: this.project.root,
+        entry: handlerInfo.entry,
+        script: undefined,
+      },
+      this.nodeConfig,
+    );
+
+    sandbox.freeze(store, 'store');
+
+    // Run handler
+    logger.debug('Running handler');
+    switch (handlerInfo.kind) {
+      case SubstrateHandlerKind.Block:
+        await sandbox.securedExec(handlerInfo.handler, [block.block]);
+        break;
+      case SubstrateHandlerKind.Call:
+        await Promise.all(
+          block.extrinsics.map(async (ext) => {
+            await sandbox.securedExec(handlerInfo.handler, [ext]);
+          }),
+        );
+        break;
+      case SubstrateHandlerKind.Event:
+        await Promise.all(
+          block.events.map(async (evt) => {
+            await sandbox.securedExec(handlerInfo.handler, [evt]);
+          }),
+        );
+        break;
+      default:
+    }
+
+    // Check expected entities
+    logger.debug('Checking expected entities');
+    let passedTests = 0;
+    let failedTests = 0;
+    for (let i = 0; i < test.expectedEntities.length; i++) {
+      const expectedEntity = test.expectedEntities[i];
+      const actualEntity = await store.get(
```

Comment:
Here we can use `expectedEntity.get(expectedEntitty.id)`

Reply:
`get` is a static method